lib/decorators/component.ts

```diff
@@ -51,9 +51,17 @@ import directiveControllerFactory from '../util/directive-controller';
 import {writeMapMulti} from './input-output';
 import {inputsMap} from '../properties/inputs-builder';
 import events from '../events/events';
+import {createConfigErrorMessage} from '../util/helpers';
 
 const TYPE = 'component';
 
+export const componentHooks = {
+	after: [],
+	extendDDO: [],
+	beforeCtrlInvoke: [],
+	afterCtrlInvoke: []
+};
+
```

Comment:
I think we should wrap the function pushing mechanism with an interface that supports type checking. That way TS warns you if your function has the wrong call type signature.

Something like:
```ts
components.addAfterHook(function(target: any, name: string, injects: any, ngModule: angular.IModule){

});
```